README.md

```diff
@@ -21,49 +22,138 @@ The payload includes the trimmed amount, together with the decimals that trimmed
-``` solidity
+
+```solidity
 emit OutboundTransferRateLimited(msg.sender, sequence, amount, getCurrentOutboundCapacity());
 ```
+
 A transfer can be released from the queue in 2 ways: (1) the capacity available exceeds the transfer amount; (2) the 24 hr period is up. In both cases, the client can call the [`completeOutboundQueuedTransfer`] function to release the transfer from the queue. The client should specify the gas amount here again to ensure that the delivery does not revert.
 
 3. Transmit
 
 Once the NttManager forwards the message to the Transceiver the message is transmitted via the [`sendMessage`] method. The method signature if enforced by the [`Transceiver`] but transceivers are free to determine their own implementation for transmitting messages.
 (e.g A message routed through the Wormhole Transceiver can be sent via automatic relaying (AR), via a specialized or custom relayer, or via the core bridge).The following event is emitted once the message has been transmitted.
-``` solidity
+
+```solidity
 emit SendTransceiverMessage(recipientChain, endpointMessage);
 ```
+
 4. Receive
 
 Once a message has been transmitted across the wire, an off-chain process (e.g. a relayer) will forward the message to the corresponding Transceiver on the recipient chain. The relayer interacts with the transceiver via an entrypoint for receiving messages (e.g. Wormhole messages are received through the [`receiveWormholeMessages`] method, which performs the messages verification along with replay protection)
 The following event is emitted during this process:
-``` solidity
+
+```solidity
 emit ReceivedRelayedMessage(deliveryHash, sourceChain, sourceAddress);
 ```
+
 This method should also forward the message to the NttManager on the recipient chain.
 NOTE: The Transceiver interface does not enforce the method signature abstractly because receiving messages may be specific to the way in which a transceiver consumes messages.
 
-``` solidity
+```solidity
 emit ReceivedMessage(vm.hash, vm.emitterChainId, vm.emitterAddress, vm.sequence);
 ```
 
 5. Attestation
 
-``` solidity
+```solidity
 emit MessageAttestedTo(digest, endpoint, _getEndpointInfosStorage()[endpoint].index);
 emit MessageAlreadyExecuted(sourceManagerAddress, digest);
 ```
 
 6. Mint or Unlock
-``` solidity
+
+```solidity
 emit TransferRedeemed(digest);
 ```
+
+### Solana
+
+1. Sending
+
+A client calls the [transfer_lock] or [transfer_burn] instruction based on whether the program is in "locking" or "burning" mode. The program mode is set during initialization. When transferring, the client must specify the amount of the transfer, the recipient chain, the recipient address on the recipient chain, and the boolean flag `should_queue` to specify whether the transfer should be queued if it hits the outbound rate limit.
```

Comment:
```suggestion
A client calls the [transfer_lock] or [transfer_burn] instruction based on whether the program is in "locking" or "burning" mode. The program mode is set during initialization. When transferring, the client must specify the amount of the transfer, the recipient chain, the recipient address on the recipient chain, and the boolean flag `should_queue` to specify whether the transfer should be queued if it hits the outbound rate limit. If `should_queue` is set to false, the transfer reverts instead of queuing if the rate limit were to be hit.
```